gapii/cc/protocol.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * GAPII-GAPIS Protocol
+ *
+ * Messages sent between GAPII and GAPIS consists of a fixed-size header
+ * followed by the message data, which can be empty.
+ *
+ * The header starts with one byte describing the message type, as defined
+ * below in the enum MessageType. It is followed by the data size, expressed
+ * as a 64bit unsigned integer, sent as 8 little-endian bytes.
```

Comment:
Should we make the size 7 bytes? :)
I mean, I don't expect us to ever send a chunk even close to 4GB, but I can get behind larger sizes...

Reply:
I would be even happy with 5 bytes. 4GB is something that could really happen, years ago I already used a 1GB 3D texture myself (texture atlas). With 5 bytes the limit would be 1TB, which should be enough for the probable lifetime of the GAPII protocol. :)

Reply:
The nice thing about 7 bytes for the size is that with the first byte for the type, we're back to 8 bytes for the (type + data size) header, which may helps with alignment etc.

Reply:
The alignment should not matter in this case. I made it 5 bytes now. Let's see if it is a good compromise. :)